buildSrc/src/main/java/org/elasticsearch/gradle/release/BreakingChangesGenerator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.release;
+
+import org.elasticsearch.gradle.Version;
+import org.elasticsearch.gradle.VersionProperties;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+public class BreakingChangesGenerator implements Closeable {
+
+    private final PrintStream out;
+
+    public BreakingChangesGenerator(File outputFile) throws FileNotFoundException {
+        this.out = new PrintStream(outputFile);
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.out.close();
+    }
+
+    public void generate(List<ChangelogEntry> entries) {
+        final Version version = VersionProperties.getElasticsearchVersion();
+
+        String majorDotMinor = version.getMajor() + "." + version.getMinor();
+        String majorMinor = String.valueOf(version.getMajor()) + version.getMinor();
+
+        List.of(
```

Comment:
I wonder if it might make sense to use some kind of templating framework here (groovy, mustache) to make updating the formatting here a bit more approachable.